### PR TITLE
Create 12.4.2 of Chef with ohai dep restriction

### PR DESCRIPTION
**Pull request by jkeiser**

### CODE REVIEW

**Comment by schisamo** [comment]
Assuming Travis is happy :+1:. I assume you'll need to push a 12.4.2 to Rubygems.org also right?

File: chef.gemspec [hunk 1]
```diff
@@ -15,13 +15,13 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = ">= 2.0.0"
 
-  s.add_dependency "chef-config", "= #{Chef::VERSION}"
+  s.add_dependency "chef-config", "= 12.4.1"
```
Comment (lamont-granquist) [inline chef.gemspec]: can't you just release chef-config 12.4.2 at the same time and avoid additional confusion here?

**Comment by lamont-granquist** [comment]
And the problem is that you need to release both the gem and the omnibus artifact or else you break people still using vagrant-omnibus and such, you can't take a shortcut around the release process.

**Comment by lamont-granquist** [comment]
(vagrant-omnibus still has some ancient code in it which pulls rubygems to get a list of valid chef versions and then will pick the latest and attempt to install it with omnibus, which will fail if you release to rubygems and do not release the omnibus artifact, its very gross, but still trolls us every time we release to rubygems ahead of the omnibus release)

**Comment by jkeiser** [comment]
@lamont-granquist what I'm hearing is that we need to release rubygems and omnibus at the same time for it to work, yes? If we do that we'll be OK?

File: chef.gemspec (hunk 1 above)
Comment (jkeiser) [inline chef.gemspec]: Sure, I can do that.

**Comment by lamont-granquist** [comment]
Correct.

**Comment by lamont-granquist** [comment]
Just do a full release of 12.4.2 and you'll be fine, don't cut corners.

**Comment by jkeiser** [comment]
No corners shall be cut :) Also, patch is updated to release chef-config 12.4.2 as well.

**Comment by lamont-granquist** [comment]
coolness

:+1:

**Comment by tyler-ball** [comment]
:+1:

**Pull request merged**